### PR TITLE
Added pagination, sorting and swagger to the project

### PR DESCRIPTION
**Pull request by kaiiiseeel**

### CODE REVIEW

**Review by AnyaRatusnyak — COMMENTED** [review]
Looks good)

File: pom.xml [hunk 1]
```diff
@@ -74,6 +74,15 @@
             <groupId>org.hibernate.validator</groupId>
             <artifactId>hibernate-validator</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-validation</artifactId>
+        </dependency>
```
Comment (andrii-hoienko) [inline pom.xml]: This is not related to task

File: src/main/java/mate/academy/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -17,36 +20,47 @@
...
     @PostMapping
+    @Operation(summary = "Create new book",
+            description = "Saves new book to the database, returns saved book")
     public BookDto createBook(@RequestBody @Valid BookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @PutMapping("/{id}")
+    @Operation(summary = "Update a book",
+            description = "Updates a specific book in database, returns the old version of book")
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/controller/BookController.java]:
```suggestion
            description = "Updates a specific book in database by id, returns the old version of book")

```

File: src/main/java/mate/academy/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -17,36 +20,47 @@
...
     public BookDto updateBookById(@PathVariable Long id,
                                   @RequestBody @Valid BookRequestDto requestDto) {
         return bookService.update(id, requestDto);
     }
 
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
+    @Operation(summary = "Delete the book",
+            description = "Deletes the specific book from database")
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/controller/BookController.java]:
```suggestion
            description = "Deletes the specific book by id from database")

```

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, left minor comments
